core/src/hlist.rs

```diff
@@ -358,7 +358,7 @@ where
 ///
 /// Users should normally allow type inference to create this type
 #[allow(dead_code)]
-pub enum Here {}
+pub struct Here;
```

Comment:
Not really, it doesn't have to be a struct. Tho it makes more sense mathematically ;)

Reply:
Ah, in that case let's have this back as a zero-member enum; it served the practical purpose of signaling that `Here` is a compile-time construct used for type-level purposes and has zero runtime cost, since it can't be instantiated.